app/domain/authentication/authn_azure/validate_application_identity.rb

```diff
@@ -0,0 +1,102 @@
+require 'command_class'
+
+module Authentication
+  module AuthnAzure
+
+    Log = LogMessages::Authentication
+    Err = Errors::Authentication::AuthnAzure
+    # Possible Errors Raised: RoleNotFound, InvalidApplicationIdentity
+
+    ValidateApplicationIdentity = CommandClass.new(
+        dependencies: {
+            role_class:                 Role,
+            resource_class:             Resource,
+            application_identity_class: ApplicationIdentity,
+            logger:                     Rails.logger
+        },
+        inputs: %i(account service_id username xms_mirid oid)
+    ) do
+
+      def call
+        parse_xms_mirid
+        # compare xms_mirid with what is defined in annotations
+        validate_application_identity
+      end
+
+      private
+
+      # validate integrity of annotations
+      def application_identity
+        @application_identity ||= @application_identity_class.new(
+            role_annotations: role_annotations,
+            service_id: @service_id
+        )
+      end
+
+      def role_annotations
+        role.annotations
+      end
+
+      # convert host ID to full host ID, cucumber:host:somepolicy
+      # TODO For a Conjur user, an incorrect full id will be returned
```

Comment:
You can remove this TODO but please add a bug for this issue as it's more general than our use-case. I can take it if you prefer.

Reply:
@orenbm please open this one 🙏

Reply:
There is actually no bug here. We can go over it tomorrow.